File: src/factories/QuestionFactory.js
class QuestionFactory {

    static createTechnology(question) {

        return {
            ...question,
            category: "technology",
            icon: "💻"
        };
    }

    static createHealth(question) {

        return {
            ...question,
            category: "health",
            icon: "🧠"
        };
    }

    static createDesign(question) {

        return {
            ...question,
            category: "design",
            icon: "🎨"
        };
    }

    static create(type, question) {

        switch (type) {

            case "technology":
                return this.createTechnology(question);

            case "health":
                return this.createHealth(question);

            case "design":
                return this.createDesign(question);

            default:
                return question;
        }
    }
}

export default QuestionFactory;

File: VocationalGuidance.API/VocationalGuidance.API/Factories/QuestionFactory.cs
using VocationalGuidance.API.Entities;

namespace VocationalGuidance.API.Factories;

public static class QuestionFactory
{
    public static Question Create(
        string text,
        string category)
    {
        return new Question
        {
            Text = text,
            Category = category
        };
    }
}